packages/fxa-settings/src/components/UnitRowSecondaryEmail/index.tsx

```diff
@@ -1,80 +1,235 @@
 /* This Source Code Form is subject to the terms of the Mozilla Public
-* License, v. 2.0. If a copy of the MPL was not distributed with this
-* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-import React, { useCallback } from 'react';
+import React, { useCallback, useRef } from 'react';
 import { useBooleanState } from 'fxa-react/lib/hooks';
+import { gql, useMutation } from '@apollo/client';
+import { useFocusOnTriggeringElementOnClose } from '../../lib/hooks';
 import UnitRow from '../UnitRow';
 import Modal from '../Modal';
 import AlertBar from '../AlertBar';
+import { useAccount, Email } from '../../models';
```

Comment:
Do we always want to be using `useAccount` in the deepest possible component?

For instance, currently we are using `useAccount` in `Settings/index.tsx` and passing in some options to `<Security/>` (and `<Profile/>` in my coming pr, which I will rebase with this after it lands).

Is the philosophy to localize the usage of `useAccount` as much as possible and attempt to _never_ pass props?

Reply:
That's my understanding, though I'm sure it wouldn't be the worst if you passed an individual prop down to a simple UI component. Maybe others have differing opinions on this.

Reply:
Yeah, since `useAccount` reads from the cache it makes sense to me to use it where we need the data. It's kind of like Context in the sense that we're pulling in those app-global values. It could make sense to pass a piece along in the props but I'd say usually it'd make more sense to just pull it in from `useAccount` where needed.

 where we can instead of drilling props down. We could pass a single prop down to a component if it made sense to, but once that component has another component...